tests/functional/gluster_stability/test_gluster_log_rotation.py

```diff
@@ -0,0 +1,102 @@
+from openshiftstoragelibs import waiter
+
+TCMU_CONF = "/etc/tcmu/tcmu.conf"
+TCMU_RUNNER_LOG = "/var/log/glusterfs/gluster-block/tcmu-runner.log"
+TCMU_ROTATE_LOG = "/etc/logrotate.d/gluster-block"
+LOG_REGEX = r"^(\d+-\d+-\d+.*\d+:\d+:\d+)\.\d+.*\[(\S+)\](.*):(\d+):(.*)$"
+
+
+class TestGlusterBlockLog(baseclass.GlusterBlockBaseClass):
+    """Class to validate logs for gluster block"""
+
+    def setUp(self):
+        super(TestGlusterBlockLog, self).setUp()
+        self.node = self.ocp_master_node[0]
+        self.g_node = self.gluster_servers[0]
+
+    def _set_log_level(self, g_node, level, msg, exec_time):
+        delete_log_level = r'sed -i "/\(^log_level.*=.*[0-9]\)/d" {}'
+        set_log_level = r'sed -i "\$alog_level = {}" {}'
+        check_log_msg = r'sed -n "/.*\({}\).*/{{p;}}" {} | tail -1'
+
+        # Set log level
+        openshift_ops.cmd_run_on_gluster_pod_or_node(
+            self.node, set_log_level.format(level, TCMU_CONF),
+            gluster_node=g_node)
+        self.addCleanup(
+            openshift_ops.cmd_run_on_gluster_pod_or_node,
+            self.node, delete_log_level.format(TCMU_CONF), gluster_node=g_node)
+
+        # Validate log level
+        log_msg = "log level now is {}".format(msg)
+        for w in waiter.Waiter(120, 3):
+            out = openshift_ops.cmd_run_on_gluster_pod_or_node(
+                self.node, check_log_msg.format(log_msg, TCMU_RUNNER_LOG),
+                gluster_node=g_node)
+            match = re.match(LOG_REGEX, out)
+            if (match
+                    and exec_time < datetime.datetime.strptime(
+                        match.group(1), "%Y-%m-%d %H:%M:%S")):
+                break
+
+        if w.expired:
+            raise exceptions.ExecutionError(
+                "Log level '{}:{}' of tcmu did not get changed on node"
+                " {}".format(level, msg, g_node))
+
+        openshift_ops.cmd_run_on_gluster_pod_or_node(
+            self.node, delete_log_level.format(TCMU_CONF), gluster_node=g_node)
+
+    @pytest.mark.tier3
+    def test_gluster_block_log_rotation(self):
+        "Verify log-rotation for gluster-block logs on independent mode"
+
+        if self.is_containerized_gluster():
+            self.skipTest(
+                "Skipping this test case as it is not independent mode setup")
+
+        get_system_time = "date '+%F %T'"
+
+        # Get current system date & time
+        exec_time = openshift_ops.cmd_run_on_gluster_pod_or_node(
+            self.node, get_system_time, gluster_node=self.g_node)
+        exec_time = datetime.datetime.strptime(exec_time, "%Y-%m-%d %H:%M:%S")
+
+        # Set log level to debug or higher level
+        self._set_log_level(self.g_node, 5, 'DEBUG SCSI CMD', exec_time)
+
+        # Get initial size on file gluster-block log
+        cmd = 'ls -lh {} | cut -d " " -f5'.format(TCMU_RUNNER_LOG)
+        initial_log_size = openshift_ops.cmd_run_on_gluster_pod_or_node(
```

Comment:
This TC is specific to independent mode cluster, so it will always have node not pod. Call function `command.cmd_run` instead.

Reply:
Done